src/pypromice/qc/persistence.py

```diff
@@ -161,3 +163,33 @@ def count_consecutive_true(
     is_first = series.astype("int").diff() == 1
     offset = (is_first * cumsum).replace(0, np.nan).fillna(method="ffill").fillna(0)
     return ((cumsum - offset + 1) * series).astype("int")
+
+
+def duration_consecutive_true(
+    series: Union[pd.Series, pd.DataFrame]
```

Comment:
It behaves differently on DataFrames. 
I'll suggest you just remove `pd.DataFrame` from accepted input types

## on `pd.Series`:
```python
> index = pd.date_range('2023-01-01 8:00', '2023-01-01 12:00', freq='10min')
> data = np.random.random(index.shape) > 0.5
> series = pd.Series(index=index, data=data)
> duration_consecutive_true(series).shape
(25,)
```

## on `pd.DataFrame`:
```
> index = pd.date_range('2023-01-01 8:00', '2023-01-01 12:00', freq='10min')
> data = np.random.random(index.shape) > 0.5
> series = pd.DataFrame(index=index, data=data)
> duration_consecutive_true(series).shape
(25, 26)
```